### PR TITLE
Remove deprecated Kubelet security controls

### LINKED ISSUE #58010: Remove deprecated Kubelet security controls in v1.13
AllowPrivileged, HostNetworkSources, HostPIDSources, HostIPCSources are deprecated in #57851 and should be removed in v1.12.

### PR DESCRIPTION
**Pull request by dims**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
 
fix: #58010 

Co-Authored-By: chaowang <chaowang95@outlook.com>

**Does this PR introduce a user-facing change?**:
```release-note
ACTION REQUIRED: Deprecated Kubelet security controls AllowPrivileged, HostNetworkSources, HostPIDSources, HostIPCSources have been removed. Enforcement of these restrictions should be done through admission control instead (e.g. PodSecurityPolicy).
ACTION REQUIRED: The deprecated Kubelet flag `--allow-privileged` has been removed. Remove any use of `--allow-privileged` from your kubelet scripts or manifests.
```

### CODE REVIEW

**Comment by charrywanganthony** [comment]
@dims Thanks, i've been away from k8s for a long time.

**Comment by charrywanganthony** [comment]
@mtaufen  PTAL.

**Comment by dims** [comment]
@liggitt @mtaufen - rebased!

**Comment by mauilion** [comment]
I was under the impression that these controls could be used to mitigate pods that were created before admission. Static pods. 
That is not the case tho. Even when `--allow-privileged=false` is set on the kubelet static pods still start up.

**Comment by dims** [comment]
@liggitt can you please approve if appropriate? (this has @mtaufen 's lgtm now)

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 	"crypto/tls"
 	"errors"
 	"fmt"
+	"k8s.io/kubernetes/pkg/capabilities"
```
Comment (liggitt) [inline cmd/kubelet/app/server.go]: import order is not correct

File: cmd/kubelet/app/server.go [hunk 2]
```diff
@@ -974,32 +974,7 @@ func RunKubelet(kubeServer *options.KubeletServer, kubeDeps *kubelet.Dependencie
...
-		return err
-	}
-
-	privilegedSources := capabilities.PrivilegedSources{
-		HostNetworkSources: hostNetworkSources,
-		HostPIDSources:     hostPIDSources,
-		HostIPCSources:     hostIPCSources,
-	}
-	capabilities.Setup(kubeServer.AllowPrivileged, privilegedSources, 0)
+	capabilities.Setup(true, capabilities.PrivilegedSources{}, 0)
```
Comment (liggitt) [inline cmd/kubelet/app/server.go]: this is the only call in the codebase to Setup(), and the middle arg is unused now. can we collapse onto `capabilities.Initialize` (and drop the PrivilegedSources field from the `capabilities.Capabilities` type)?

**Review by liggitt — COMMENTED** [review]
a couple nits, just to avoid leaving dead code in place

**Comment by liggitt** [comment]
/approve

/hold
for last cleanup comments

File: cmd/kubelet/app/server.go (hunk 1 above)
Comment (dims) [reply cmd/kubelet/app/server.go]: Done

File: cmd/kubelet/app/server.go (hunk 2 above)
Comment (dims) [reply cmd/kubelet/app/server.go]: switching from capabilities.Setup to capabilities.Initialize

**Comment by dims** [comment]
/hold cancel

done with "cleanup comments"

**Pull request merged**